great_expectations/datasource/data_connector/inferred_asset_sql_data_connector.py

```diff
@@ -264,6 +264,30 @@ def _introspect_db(
                             }
                         )
 
+        # SQLAlchemy's introspection does not list "external tables" in Redshift Spectrum (tables whose data is stored on S3).
+        # The following code fetches the names of external schemas and tables from a special table
+        # 'svv_external_tables'.
+        try:
+            if "redshift" == engine.dialect.name.lower():
```

Comment:
This may need to include "redshift+psycopg2" as dialect

Reply:
@rreinoldsc I grepped the code and and see that "redshift" is the dialect name we are using in all the other cases when we check if the database is Redshift